index.js

```diff
@@ -1492,23 +1503,27 @@
 
   //. ### Composition
 
-  //# compose :: (b -> c) -> (a -> b) -> a -> c
+  //# compose :: Semigroupoid s => s b c -> s a b -> s a c
   //.
-  //. Composes two unary functions, from right to left. Equivalent to Haskell's
-  //. `(.)` function.
+  //. Curried version of [`Z.compose`][].
   //.
-  //. This is the B combinator from combinatory logic.
+  //. When specialized to Function, `compose` composes two unary functions,
+  //. from right to left (this is the B combinator from combinatory logic).
+  //.
+  //. The generalized type signature indicates that `compose` is compatible
+  //. with any [Semigroupoid][].
   //.
   //. See also [`pipe`](#pipe).
   //.
   //. ```javascript
-  //. > S.compose(Math.sqrt, S.add(1), 99)
+  //. > S.compose(Math.sqrt, S.add(1))(99)
   //. 10
   //. ```
-  function compose(f, g, x) {
-    return f(g(x));
-  }
-  S.compose = def('compose', {}, [Fn(b, c), Fn(a, b), a, c], compose);
+  S.compose =
+  def('compose',
+      {s: [Z.Semigroupoid]},
+      [s(b, c), s(a, b), s(a, c)],
+      Z.compose);
```

Comment:
This is a breaking change. I didn't realize the function is currently ternary. It was surprising.